builtin/logical/pki/path_roles.go

```diff
@@ -644,6 +645,10 @@ func (b *backend) pathRoleCreate(ctx context.Context, req *logical.Request, data
 	// no_store implies generate_lease := false
 	if entry.NoStore {
 		*entry.GenerateLease = false
+		if data.Get("generate_lease").(bool) {
```

Comment:
Would it make sense to return an error when both options are set to true?

Reply:
The [docs](https://github.com/hashicorp/vault/blob/main/builtin/logical/pki/path_roles.go#L370-L371) have mentioned [since the introduction of the `no_store` option](https://github.com/hashicorp/vault/commit/7cbc5d6e058647ea45f3254def7ffff060b59faa#diff-dae4006668fecb2f5554c7b3efa6f21df59a7055280dba36d435251372f0076cR215) that `no_store=true` sets `generate_lease=false` (regardless of its value), so I don't think we can necessarily make a breaking change at this point in time (which erring out would be). However, emitting a warning is the next best thing we can do for the time being. :-)

A future Vault v2 version potentially could make that change, or perhaps an equivalent amount of time with the warning. I dunno. But at least as it stands, we'd risk breaking people's existing e.g., Terraform definitions if we made that change now.

Reply:
The nice thing is a [quick search](https://github.com/search?p=1&q=vault_pki_secret_backend_role+generate_lease+no_store&type=Code) shows that no _public_ Terraform users seem to have made this mistake, but [some use variables](https://github.com/adfinis-sygroup/terraform-vault-namespace/blob/ef2987f2efea2da331f9f9bc94bbc2497ff92db7/variables.tf#L168-L186) that could potentially be overridden to have the undesired behavior. Plus, I'd venture that most Terraform users keep their stuff private.